### PR TITLE
added new playbook validation for deletecontext all=yes

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29836

## Description
Validate fails if there is a usage of DeleteContext in a playbook using all=yes

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by roysagi** [comment]
@orhovy - Please add a changelog to this.
Also should open an issue about this so it would be documented.

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -732,6 +733,11 @@ def playbook_unhandled_condition(task_id, task_condition_labels):
     def playbook_unconnected_tasks(orphan_tasks):
         return f'The following tasks ids have no previous tasks: {orphan_tasks}'
 
+    @staticmethod
+    @error_code_decorator
+    def playbook_can_have_deletecontext_all():
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:738]:
```suggestion
    def playbook_cant_have_deletecontext_all():
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -109,6 +109,7 @@
     "playbook_unhandled_condition": "PB102",
     "playbook_unconnected_tasks": "PB103",
     "invalid_deprecated_playbook": "PB104",
+    "playbook_can_have_deletecontext_all": "PB105",
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:112]:
```suggestion
    "playbook_cant_have_deletecontext_all": "PB105",
```

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 3]
```diff
@@ -253,3 +255,21 @@ def is_valid_deprecated_playbook(self) -> bool:
...
+        Returns:
+            True if delete context exists else False.
+        """
+        tasks: Dict = self.current_file.get('tasks', {})
+        for task in tasks.values():
+            curr_task = task.get('task', {})
+            scriptargs = task.get('scriptarguments', {})
+            if curr_task and scriptargs and curr_task.get('scriptName', '') == 'DeleteContext' \
+                    and scriptargs.get('all', {}).get('simple', '') == 'yes':
+                error_message, error_code = Errors.playbook_can_have_deletecontext_all()
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/playbook.py:271]:
```suggestion
                error_message, error_code = Errors.playbook_cant_have_deletecontext_all()
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**